homeassistant/components/mill/climate.py

```diff
@@ -111,23 +108,13 @@ def __init__(
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, heater.device_id)},
             manufacturer=MANUFACTURER,
-            model=f"Generation {heater.generation}",
+            model=heater.model,
             name=heater.name,
         )
-        if heater.is_gen1:
-            self._attr_hvac_modes = [HVACMode.HEAT]
-        else:
-            self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
-
-        if heater.generation < 3:
-            self._attr_supported_features = (
-                ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
-            )
-            self._attr_target_temperature_step = PRECISION_WHOLE
-        else:
-            self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
-            self._attr_target_temperature_step = PRECISION_HALVES
+        self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
 
+        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+        self._attr_target_temperature_step = PRECISION_TENTHS
```

Comment:
These can be initialized outside of the `__init__` as they don't change

Reply:
Still a comment, `_attr_hvac_modes`, `_attr_supported_features` and `_attr_target_temperature_step` can be set outside of the `__init__`

Reply:
No worries :)